integration/query_fuzz_test.go

```diff
@@ -970,15 +971,34 @@ var comparer = cmp.Comparer(func(x, y model.Value) bool {
 		return l == r || (compareFloats(float64(l.Count), float64(r.Count)) && compareFloats(float64(l.Sum), float64(r.Sum)) && compareHistogramBuckets(l.Buckets, r.Buckets))
 	}
 
+	fetchValuesFromNH := func(nhString string) []float64 {
+		// Regex to match float numbers
+		re := regexp.MustCompile(`-?\d+(\.\d+)?`)
```

Comment:
Instead of using a regex here can we use parser.ParseSeriesDesc() to parse this into a real histogram? 

See: https://github.com/prometheus/prometheus/blob/main/promql/parser/parse.go#L252

```
value := r.Metric.Get("value")
lbls, val, err := parser.ParseSeriesDesc(value)
```

This worked for me.

Reply:
Probably `parser.ParseSeriesDesc` doesn't support NH format now. We can go ahead with the regex I think

Reply:
You're right. I was mistaken earlier.